### PR TITLE
[CI] Move from EndBug/add-and-commit to stefanzweifel/git-auto-commit-action

### LINKED ISSUE #7579: Update from EndBug/add-and-commit to stefanzweifel/git-auto-commit-action
# Feature Request

For rectify auto-commit, we currently only can use `EndBug/add-and-commit@v7.5.0` due to error after v8+

In current used version, it seems show notice:

```bash
Run EndBug/add-and-commit@v7.5.0
  
Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

We seems possibly can update to use stefanzweifel/git-auto-commit-action https://github.com/stefanzweifel/git-auto-commit-action which seems there is already handle for it:

- https://github.com/stefanzweifel/git-auto-commit-action/pull/252

### PR DESCRIPTION
**Pull request by samsonasik**
Reopen of PR:

- https://github.com/rectorphp/rector-src/pull/3168

that was closed and got overlapped commit so new PR is needed.

Fixes https://github.com/rectorphp/rector/issues/7579

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba the workflows got pending:

```
rector (config utils) Expected — Waiting for status to be reported
Required
rector (rules) Expected — Waiting for status to be reported
Required
rector (src tests rules-tests packages packages-tests) Expected — Waiting for status to be reported
Required 
```

could you verify if it need manual enable?

**Comment by TomasVotruba** [comment]
These should work as before.

Seems something wrong the workflow syntax.

File: .github/workflows/rector.yaml [hunk 1]
```diff
@@ -64,17 +64,13 @@ jobs:
 
             -   run: vendor/bin/ecs check --fix --ansi
 
-            # see https://github.com/EndBug/add-and-commit
+            # see https://github.com/rectorphp/rector/issues/7579
             -
                 # commit only to core contributors who have repository access
                 if: github.event.pull_request.head.repo.full_name == github.repository
-                uses: EndBug/add-and-commit@v7.5.0
```
Comment (TomasVotruba) [inline .github/workflows/rector.yaml:47]: Would new https://github.com/EndBug/add-and-commit 9.1 help?

Comment (samsonasik) [reply .github/workflows/rector.yaml:47]: that not work, see dependabot PR: https://github.com/rectorphp/rector-src/pull/3026

File: .github/workflows/rector.yaml [hunk 2]
```diff
@@ -72,4 +72,3 @@ jobs:
                 with:
                     commit_message: [ci-review] Rector Rectify
                     commit_author: GitHub Action <actions@github.com>
-                    commit_user_email: action@github.com
```
Comment (TomasVotruba) [inline .github/workflows/rector.yaml]: These values might need quoting, as they contain weird characters

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Missclick on the approve, just wanted to send one comment :)

File: .github/workflows/rector.yaml (hunk 2 above)
Comment (samsonasik) [reply .github/workflows/rector.yaml]: It works 🎉 https://github.com/rectorphp/rector-src/pull/3426/commits/4fde336cf238bc44373c5fab2fbcb13be095c58b

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba the commit is applied as expected https://github.com/rectorphp/rector-src/pull/3426/commits/4fde336cf238bc44373c5fab2fbcb13be095c58b, I think it is ready.

**Comment by samsonasik** [comment]
Rebased.

**Comment by TomasVotruba** [comment]
Thanks 🙏

Does the CI Rector commit triggers the whole workflow?
I can't see it after the rebase. That's very important part

**Comment by samsonasik** [comment]
Yes, here https://github.com/rectorphp/rector-src/pull/3426/commits/d84398b7b3c4d4555c5178cfa153238ab933e3fa, I will re-add unused property.

**Comment by samsonasik** [comment]
Re-added unused property https://github.com/rectorphp/rector-src/pull/3426/commits/b691b660700743692ac048e67b252af6b06ee304

**Comment by samsonasik** [comment]
Here above trigger commit https://github.com/rectorphp/rector-src/pull/3426/commits/86fa5454038e4c46e7422419ab6c8180a9f6c7c9 on Rectify 👍

**Comment by TomasVotruba** [comment]
Very nice 👍🙂 thank you

**Pull request merged**